c/logging.c

```diff
@@ -647,8 +652,118 @@ int logGetLevel(LoggingContext *context, uint64 compID){
   return component ? component->currentDetailLevel : ZOWE_LOG_NA;
 }
 
-void zowelog(LoggingContext *context, uint64 compID, int level, char *formatString, ...){
+// creates a formatted timestamp to be used in zowe log
+static void getLogTimestamp(char *date) {
+   int64 stck = 0;      
+   char timestamp[14]; // unformated time
+   uint32 out_secs = 0;
+   uint32 out_micros = 0;
+
+   getSTCK(&stck);
+   int64 unixTime = stckToUnix(stck); // gets unixTime
+   stckToUnixSecondsAndMicros(stck, &out_secs, &out_micros); // calculates microseconds
+   out_micros = out_micros/1000;
+   unixToTimestamp((uint64) unixTime, timestamp);
+
+   // format: 2019-03-19 11:23:57.776
+   snprintf(date,24, "%.4s-%.2s-%.2s %.2s:%.2s:%.2s.%.3d"
+      ,timestamp, timestamp+4,timestamp+6,timestamp+8,timestamp+10,timestamp+12,(unsigned int) out_micros);
```

Comment:
It's bad practice to hard code offsets or lengths. If you need reformat a data format I suggest using a struct and/or a new function to do that formatting like `strftime()`.